alacritty_terminal/src/selection.rs

```diff
@@ -66,6 +66,15 @@ impl<L> SelectionRange<L> {
     }
 }
 
+/// Different kinds of selection.
+#[derive(Debug, Clone, PartialEq)]
+pub enum SelectionType {
+    Simple,
+    Block,
+    Semantic,
+    Lines,
+}
```

Comment:
Shouldn't be relevant since it is never attempted to copy/clone, but seems easy enough to add.

I've also changed things to remove `pub` here, since it's never used publicly.